local_planner/test/test_planner_functions.cpp

```diff
@@ -78,3 +79,556 @@ TEST(PlannerFunctions, generateNewHistogramSpecificCells) {
+                     path_selected, path_rejected, path_blocked, path_waypoints,
+                     cost_path_candidates, goal, position, position_old,
+                     goal_cost_param, smooth_cost_param,
+                     height_change_cost_param_adapted, height_change_cost_param,
+                     only_yawed, resolution_alpha);
+
+  // THEN: all directions should be classified as path_candidates
+  EXPECT_EQ(
+      std::floor(360 / resolution_alpha) * std::floor(180 / resolution_alpha),
+      path_candidates.cells.size());
+  EXPECT_EQ(0, path_rejected.cells.size());
+  EXPECT_EQ(0, path_blocked.cells.size());
+}
+
+TEST(PlannerFunctionsTests, findFreeDirectionsNoWrap) {
+  // GIVEN: an histogram with a rejected cell, no wrapping of blocked cells
+  int resolution_alpha = 5;
+  double safety_radius = 5.0;
+  Histogram histogram = Histogram(resolution_alpha);
+  int obstacle_idx_e = 5;
+  int obstacle_idx_z = 10;
+  histogram.set_bin(obstacle_idx_e, obstacle_idx_z,
+                    histogram.get_bin(obstacle_idx_e, obstacle_idx_z) + 1);
+  // all the variables below aren't used by the test
+  geometry_msgs::Point goal;
+  goal.x = 2.0f;
+  goal.y = 0.0f;
+  goal.z = 2.0f;
+  geometry_msgs::PoseStamped position;
+  position.pose.position.x = 0.0f;
+  position.pose.position.y = 0.0f;
+  position.pose.position.z = 2.0f;
+  geometry_msgs::Point position_old;
+  position_old.x = -1.0f;
+  position_old.y = 0.0;
+  position_old.z = 2.0f;
+  double goal_cost_param = 2.0;
+  double smooth_cost_param = 1.5;
+  double height_change_cost_param_adapted = 4.0;
+  double height_change_cost_param = 4.0;
+  bool only_yawed = false;
+
+  // expected output
+  std::vector<std::pair<int, int>> blocked;
+  std::vector<std::pair<int, int>> free;
+  for (int e = 0; e < (180 / resolution_alpha); e++) {
+    for (int z = 0; z < (360 / resolution_alpha); z++) {
+      if (!(z <= 11 && z >= 9 && e >= 4 && e <= 6)) {
+        free.push_back(std::make_pair(e, z));
+      } else {
+        if (!(e == obstacle_idx_e && z == obstacle_idx_z)) {
+          blocked.push_back(std::make_pair(e, z));
+        }
+      }
+    }
+  }
+
+  // WHEN: we look for free directions
+  nav_msgs::GridCells path_candidates, path_selected, path_rejected,
+      path_blocked, path_waypoints;
+  std::vector<float> cost_path_candidates;  // not needed
+
+  findFreeDirections(histogram, safety_radius, path_candidates, path_selected,
+                     path_rejected, path_blocked, path_waypoints,
+                     cost_path_candidates, goal, position, position_old,
+                     goal_cost_param, smooth_cost_param,
+                     height_change_cost_param_adapted, height_change_cost_param,
+                     only_yawed, resolution_alpha);
+
+  // THEN: we should have one rejected cell and the 8 neightbooring cells
+  // blocked
+  for (int i = 0, j = 0; i < path_blocked.cells.size(), j < blocked.size();
```

Comment:
These test blocks seem to be used a lot, maybe they could be put in a function at the top of the file?